src/binary-reader-ir.cc

```diff
@@ -843,28 +863,38 @@ Result BinaryReaderIR::OnLocalNameLocalCount(Index index, Index count) {
 }
 
 Result BinaryReaderIR::OnInitExprF32ConstExpr(Index index, uint32_t value) {
-  *current_init_expr = Expr::CreateConst(Const(Const::F32(), value));
+  Expr* expr = Expr::CreateConst(Const(Const::F32(), value, GetLocation()));
+  expr->loc = GetLocation();
```

Comment:
This one assigns the Expr::loc, the other one assigns the Const::loc. In the text format these can be different. But honestly, it would be nicer to pass the location in to `Expr::CreateConst` directly, I just can't do it with a default parameter until I add a `Location` default constructor. This is currently hard to do because `Location` is used in places where it assumes it can have a trivial constructor. I could fix that here, but I figured it would be better to do in another PR.